packages/web-components/src/search/search.styles.ts

```diff
@@ -0,0 +1,145 @@
+import { css, ElementStyles } from '@microsoft/fast-element';
+import {
+  Button,
+  display,
+  ElementDefinitionContext,
+  forcedColorsStylesheetBehavior,
+  FoundationElementDefinition,
+} from '@microsoft/fast-foundation';
+import {
+  inputFilledForcedColorStyles,
+  inputFilledStyles,
+  inputForcedColorStyles,
+  inputStateStyles,
+  inputStyles,
+} from '../styles';
+import { appearanceBehavior } from '../utilities/behaviors';
+import { designUnit } from '../design-tokens';
+import { DirectionalStyleSheetBehavior } from '../styles';
+
+/**
+ * LTR styles for calendar
+ * @internal
+ */
+ const ltrStyles = css`
+ .clear-button {
+   right: 1px;
+ }
+ `;
+
+ /**
+  * RTL styles for calendar
+  * @internal
+  */
+ const rtlStyles = css`
+ .clear-button {
+  left: 1px;
+ }
+ `;
+
+export const searchFilledStyles: (
+  context: ElementDefinitionContext,
+  definition: FoundationElementDefinition,
+) => ElementStyles = (context: ElementDefinitionContext, definition: FoundationElementDefinition) =>
+  css`
+    ${inputFilledStyles(context, definition, '.root')}
+  `.withBehaviors(
+    forcedColorsStylesheetBehavior(
+      css`
+        ${inputFilledForcedColorStyles(context, definition, '.root')}
+      `,
+    ),
+  );
+
+export const searchStyles = (context, definition) =>
+  css`
+    ${display('inline-block')}
+
+    ${inputStyles(context, definition, '.root')}
+
+    ${inputStateStyles(context, definition, '.root')}
+
+    .root {
+      display: flex;
+      flex-direction: row;
+    }
+
+    .control {
+      -webkit-appearance: none;
+      color: inherit;
+      background: transparent;
+      border: 0;
+      height: calc(100% - 4px);
+      margin-top: auto;
+      margin-bottom: auto;
+      padding: 0 calc(${designUnit} * 2px + 1px);
+      font-family: inherit;
+      font-size: inherit;
+      line-height: inherit;
+    }
+
+    .clear-button {
```

Comment:
We should avoid absolute position here (and the need for custom rtl above). If the `.root` container is flex, why doesn't this work inline with the other elements? This doesn't seem like it will scale well for customizations like density.

Reply:
Because clear button is not only visible, we we inject it into the `flex` layout the component will shift its width. Its the same way in `fast-components`

Reply:
Isn't the width set on the container element, so the visible width of the control would be constant but whether the clear button is there or not would only change the width of the inner input element, which doesn't have a visible border?

I'm actually dealing with something similar in density work right now. Is this perhaps to get around the padding internal to the root element?

Reply:
If it's the latter example, needing to overlap the internal padding on the `root` frame, I've been looking at solutions, and I know there is intrinsic distrust of them, but one of the ways I've found is for negative margins on the button. This is the best solution I've found at this point because it maintains the button's design decisions (density, etc) without forcing the height and relying on top/right/bottom/left offsets.

The other solution is to place padding on the elements that need it. In this case the button would opt-out, nestling it into the end. The actual `input` and the icon on the left would use `margin` to keep themselves away from the sides.

Neither are perfect. This type of layout is difficult with css peculiarities.

Reply:
Removed `absolute` position, seems to render the same. Didn't have to do anything fancy. I think I had it this way because of an older version of the template structure, but now they input elements are wrapped in a flex box it just works.